test/README.md

```diff
@@ -1,18 +1,97 @@
 # Running Tests
 
-To run the tests, perform the following steps:
+To run all tests, perform the following steps:
 
-1) Startup a fresh iRODS server.
-2) Startup the Testing keycloak image.
-3) Startup the iRODS HTTP API server.
-4) Ensure the HTTP API can communicate with the iRODS server and the keycloak image.
-5) Configure the [config.py](config.py) to the appropriate test configuration.
-6) Install `pytest` for the OpenID tests.
-7) Run `pytest` to run all tests. If desired, run `test_irods_http_api.py` directly if OpenID tests are not required.
+## Startup a fresh iRODS server
 
-## Testing support for OpenID Connect via Keycloak
+Use the iRODS testing environment to setup an iRODS server: https://github.com/irods/irods_testing_environment
+
+### Get the iRODS network information
+
+We need the network that the HTTP API is running on to have all pieces of the server
+communicate with each other.
+
+Following is an example of how to list all available docker networks with examples
+of how the output may appear:
+
+```console
+$ docker network ls
+NETWORK ID     NAME                              DRIVER    SCOPE
+d06849108e9c   bridge                            bridge    local
+371117285a13   host                              host      local
+fbf586b6459f   none                              null      local
+297992bf57a9   ubuntu-2204-postgres-14_default   bridge    local
+```
+
+In the example output, the desired network would be `ubuntu-2204-postgres-14_default`.
+
+The docker network used will be referred to as `<NETWORK>` in the following sections.
+
+## Startup the Keycloak image for testing
+
+See [Build the Testing Image](#build-the-testing-image) to build the image first.
+
+After which, run the following command to startup keycloak:
+
+```console
+$ docker run --rm --network <NETWORK> -p 8080:8080 irods-http-api-keycloak start-dev
+```
+
+### Get detailed network information
+Detailed information of the network will be required to properly configure the HTTP API.
+
+To get this information run the following command:
+```console
+$ docker network inspect <NETWORK>
+```
```

Comment:
What does this step do? Do we need to use the information somewhere? Sorry if I missed it...

Reply:
A lot of the subsections are placed a bit oddly, but I try to put them before the section they are used it.

This specifically is used for creating a valid HTTP API configuration in the following section.

Reply:
Okay. I guess it wasn't clear to me what specifically is used in the configuration and how, which is where my confusion came from. Perhaps more words would be helpful here, but I'll defer at this point.

Reply:
Hmm, maybe a brief statement like `For example, <THING> requries network infromation from this.`.

Not the exact wording I'll use, but the general idea.

Reply:
Done.